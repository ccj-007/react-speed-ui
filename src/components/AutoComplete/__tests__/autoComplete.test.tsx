import React from "react";
import { render, fireEvent, screen, waitFor } from "@testing-library/react";
import { AutoComplete } from "../../index";
import '@testing-library/jest-dom/extend-expect';

describe('test autoComplete component', () => {
  it('should render', async () => {
    let lakers = [
      { value: "aaa", name: "我的小 A" },
      { value: "aaaaaaaa我是小A", name: "我的小 B" },
    ];
    const Demo = () => {
      return (
        <AutoComplete
          onSelect={() => { }}
          fetchSuggestions={lakers}
          width={600}
        ></AutoComplete>
      )
    }
    const { asFragment, getByTestId } = render(<Demo />)
    expect(asFragment()).toMatchSnapshot()

    fireEvent.change(getByTestId('input'), { target: { value: 'a' } })
    fireEvent.click(getByTestId('input'))

    expect(getByTestId('input')).toHaveAttribute('value', 'a')

    await waitFor(() => {
      expect(screen.getByText(/aaa/i)).toBeInTheDocument()
    }, { timeout: 5000 })
  })
})