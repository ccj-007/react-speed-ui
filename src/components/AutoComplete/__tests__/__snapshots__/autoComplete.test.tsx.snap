// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test autoComplete component should render 1`] = `
<DocumentFragment>
  <div
    class="speed-autoComplete"
  >
    <div
      class="speed-input-warp"
      style="width: 600px;"
    >
      <input
        class="speed-input"
        data-testid="input"
        type="text"
        value=""
      />
    </div>
    <div
      class=""
      style="width: 600px;"
    />
  </div>
</DocumentFragment>
`;
